tests/bgp/test_traffic_shift.py

```diff
@@ -0,0 +1,275 @@
+import pytest
+import logging
+import json
+import time
+import ipaddr as ipaddress
+from tests.common.helpers.assertions import pytest_assert
+from tests.common.utilities import wait_tcp_connection
+from jinja2 import Template
+import re
+
+pytestmark = [
+    pytest.mark.topology('t1')
+]
+
+logger = logging.getLogger(__name__)
+
+TS_NORMAL = "System Mode: Normal"
+TS_MAINTENANCE = "System Mode: Maintenance"
+TS_ERROR = "System Mode: Not consistent"
+
+DUMP_FILE = "/tmp/bgp_monitor_dump.log"
+CUSTOM_DUMP_SCRIPT = "bgp/bgp_monitor_dump.py"
+CUSTOM_DUMP_SCRIPT_DEST = "/usr/share/exabgp/bgp_monitor_dump.py"
+BGP_MONITOR_PORT = 7000
+BGP_MONITOR_NAME = "bgp_monitor"
+BGP_ANNOUNCE_TIME = 30 #should be enough to receive and parse bgp updates
+
+# TODO: remove me
+BGPMON_TEMPLATE_FILE = 'bgp/templates/bgpmon.j2'
+BGPMON_CONFIG_FILE = '/tmp/bgpmon.json'
+
+@pytest.fixture
+def traffic_shift_community(duthost):
+    community = duthost.shell('sonic-cfggen -y /etc/sonic/constants.yml -v constants.bgp.traffic_shift_community')['stdout']
+    return community
+
+@pytest.fixture
+def common_setup_teardown(ptfhost, duthost, localhost):
+    mg_facts = duthost.minigraph_facts(host=duthost.hostname)['ansible_facts']
+    dut_lo_addr = mg_facts['minigraph_lo_interfaces'][0]['addr']
+    dut_mgmt_ip = mg_facts['minigraph_mgmt_interface']['addr']
+    asn = mg_facts['minigraph_bgp_asn']
+    peer_addr = ptfhost.mgmt_ip
+    # TODO: Add a common method to load BGPMON config for test_bgpmon and test_traffic_shift
```

Comment:
another TODO? :)

Reply:
Yes. I will check if we could share some code or method among these test cases after they are merged, such as start bgpmon. So the TODOs are left here.